config/v1/types_dns.go

```diff
@@ -74,6 +80,16 @@ type DNSZone struct {
 	Tags map[string]string `json:"tags,omitempty"`
 }
 
+// Upstream defines the schema for associating a base domain with a list of name resolvers.
+type Upstream struct {
+	// domain is the base domain to match to forward name queries.
+	Domain string `json:"domain"`
+	// resolverList is a list of name servers responsible for resolving name queries from Domain.
+	// Each resolver in the list is represented by an IP address or IP:port if the resolver
+	// listens on a UDP port other than 53.
+	ResolverList []string `json:"resolverList"`
```

Comment:
Another possible name for the field would be `Nameservers` — the List suffix seems redundant in any case. `Resolvers`?

Reply:
+1 for `Nameservers`.